yt_dlp_plugins/postprocessor/stash.py

```diff
@@ -72,4 +79,214 @@ def run(self, info):
+            self.to_screen("[Debug] Initiating metadata scan on path: " + dirpath)
+            stash_meta_job = self.stash.metadata_scan(paths=dirpath, flags={
+                "scanGenerateCovers": False,
+                "scanGeneratePreviews": False,
+                "scanGenerateImagePreviews": False,
+                "scanGenerateSprites": False,
+                "scanGeneratePhashes": False,
+                "scanGenerateThumbnails": False,
+                "scanGenerateClipPreviews": False
+            })
+            self.to_screen(f"[Debug] Metadata scan job ID: {stash_meta_job}")
+
+            # Step 4: Wait until metadata scan is complete
+            while True:
+                job_status = self.stash.find_job(stash_meta_job)
+                self.to_screen(f"[Debug] Metadata scan job status: {job_status}")
+                if job_status["status"] == "FINISHED":
+                    break
+                elif job_status["status"] == "FAILED":
+                    self.to_screen("[Error] Metadata scan job failed.")
+                    return [], info
+                sleep(0.5)
+
+            # Step 5: Find the newly created scene
+            self.to_screen(f"[Debug] Looking for scene with path: {filepath}")
+            scene = self.stash.find_scenes({"path": {"modifier": "EQUALS", "value": filepath}})
+            self.to_screen(f"[Debug] Scene search result: {scene}")
+            if not scene or len(scene) == 0:
+                self.to_screen("[Error] No scene found after metadata scan. Please verify the path and metadata settings.")
+                return [], info
+
+            scene_id = scene[0]["id"]
+            self.to_screen(f"[Debug] Found scene with id: {scene_id}")
+
+            # Step 6: Scrape metadata from URL
+            if "webpage_url" not in info:
+                self.to_screen("[Error] No URL found for scraping")
+                return [], info
+
+            self.to_screen(f"[Debug] Scraping metadata from URL: {info['webpage_url']}")
+            scene_data = self.scrape_scene_by_url(info['webpage_url'])
+
+            if not scene_data:
+                self.to_screen("[Error] Error or no data found during scraping.")
+                return [], info
+
+            # Step 7: Update the scene data using the scraped information
+            update_scene = {
+                "id": scene_id,
+                "url": info.get("webpage_url"),
+            }
+
+            if scene_data.get("title"):
+                update_scene["title"] = scene_data["title"]
+            elif info.get("title"):
+                update_scene["title"] = info["title"]
+
+            if scene_data.get("details"):
+                update_scene["details"] = scene_data["details"]
+            elif info.get("description"):
+                update_scene["details"] = info["description"]
+
+            if scene_data.get("date"):
+                update_scene["date"] = scene_data["date"]

```

Comment:
Why not fallback to `info.get("upload_date")` here?
might even make sense to add a time parser helper function that can be used in the native scrape.
```suggestion
            if scene_data.get("date"):
                update_scene["date"] = scene_data["date"]
            elif info.get("upload_date"):
                update_scene["date"] = (
                    info["upload_date"][0:4]
                    + "-"
                    + info["upload_date"][4:6]
                    + "-"
                    + info["upload_date"][6:8]
                )
```